### ISSUE TITLE
10$: Error on invalid TypeScript 'export default'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#796]
The following export statements are illegal in TypeScript. We should report a nice diagnostic.
```typescript
export default namespace ns {}  // error
export default enum e {}  // error
export default type T = any;  // error
export default import A = ns.B;  // error
```

**Comment by strager** [comment]
Similar diagnostics which already exist in quick-lint-js:
```js
// E0076: cannot declare and export variable with 'export default'             
export default const x = 42;
// E0076: cannot declare and export variable with 'export default'             
export default let x = 42;
// E0076: cannot declare and export variable with 'export default'             
export default var x = 42;
```
